examples/java/JavaExample.java

```diff
@@ -599,7 +599,7 @@ void quantifierExample1(Context ctx)
         System.out.println("QuantifierExample");
         Log.append("QuantifierExample");
 
-        Sort[] types = new Sort[3];
+        IntSort[] types = new IntSort[3];
```

Comment:
Change is necessary because of cast to ArithExpr<?>, which is technically not strictly compatible with Expr<?> because ArithExpr<T extends ArithSort> specifies that the generic T is more specific than Sort. There are ways around this but it makes things a little clunky and removes the advantages that typing like this poses.

Provided that most users are specifically typing, this will likely not affect many users.